### PR TITLE
Backport vote extension metrics

### PR DESCRIPTION
**Pull request by sergio-mena**
Contributes to #10

Partial cherry-pick of #8480

The rest of #8480 was backported for `v0.37.x`.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: docs/core/metrics.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ The following metrics are available:
 | consensus\_block\_gossip\_parts\_received  | Counter   | matches\_current | Number of block parts received by the node                                                                                                 |
 | consensus\_quorum\_prevote\_delay          | Gauge     |                  | Interval in seconds between the proposal timestamp and the timestamp of the earliest prevote that achieved a quorum                        |
 | consensus\_full\_prevote\_delay            | Gauge     |                  | Interval in seconds between the proposal timestamp and the timestamp of the latest prevote in a round where all validators voted           |
+| consensus\_vote\_extension\_receive\_count | Counter   | status           | Number of vote extensions received                                                                                                         |
```
Comment (jmalicevic) [inline docs/core/metrics.md:47]: This is just a backport so it's fine but to mee somehow this metric is showing two things (number and status) and I am not sure whether one could extract stats on the number of rejects or accepts. Maybe it is not even relevant, but was a bit counterintuitive at first.

File: consensus/metrics.go [hunk 2]
```diff
@@ -138,6 +153,14 @@ func (m *Metrics) MarkRound(r int32, st time.Time) {
...
 	roundTime := time.Since(st).Seconds()
 	m.RoundDurationSeconds.Observe(roundTime)
+
+	pvt := cmtproto.PrevoteType
+	pvn := strings.ToLower(strings.TrimPrefix(pvt.String(), "SIGNED_MSG_TYPE_"))
+	m.RoundVotingPowerPercent.With("vote_type", pvn).Set(0)
+
+	pct := cmtproto.PrecommitType
+	pcn := strings.ToLower(strings.TrimPrefix(pct.String(), "SIGNED_MSG_TYPE_"))
+	m.RoundVotingPowerPercent.With("vote_type", pcn).Set(0)
```
Comment (jmalicevic) [inline consensus/metrics.go:163]: How are these changes related to vote extensions and why is the count here set to 0 - because we start a new round?

Comment (sergio-mena) [reply consensus/metrics.go:163]: The original patch (the one I'm cherry-picking from) was not limited to vote extensions. This hunk is a left-over that hadn't been backported from the original patch.
This is the typical example why we're following this process (i.e., cherry-picking patches that are supposed to have already been backported manually)

File: docs/core/metrics.md (hunk 1 above)
Comment (sergio-mena) [reply docs/core/metrics.md:47]: My interpretation is that you want to count both, accepted and rejected, as two "sub-metrics" (that's my understanding of `MarkVoteExtensionReceived`

Comment (jmalicevic) [reply docs/core/metrics.md:47]: You should definitely count both towards the number of VE seen. I was just wondering whether someone could further count the number within each sub-metric. But not really a major issue, just discussing.

**Review by jmalicevic — APPROVED** [review]
